### PR TITLE
Add support for SECURITY LABEL on ROLE propagation from non-main databases

### PR DESCRIPTION
**Pull request by eaydingol**
DESCRIPTION: Adds support for distributed "SECURITY LABEL on ROLE" commands from the databases where Citus is not installed.

### CODE REVIEW

File: src/test/regress/sql/seclabel_non_maindb.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+SET citus.enable_create_database_propagation to ON;
+
+CREATE DATABASE database1;
+CREATE DATABASE database2;
+
+\c - - - :worker_1_port
+SET citus.enable_create_database_propagation to ON;
+CREATE DATABASE database_w1;
+
+
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel_non_maindb.sql:14]: Given that SET commands only alters the value of the GUC for the current session (and \c means switching into a different connection/session), those two lines don't have any affect for the rest of the tests.

```suggestion
```

File: src/test/regress/multi_schedule [hunk 2]
```diff
@@ -109,6 +109,7 @@ test: undistribute_table
 test: run_command_on_all_nodes
 test: background_task_queue_monitor
 test: other_databases grant_role_from_non_maindb
+test: seclabel_non_maindb
```
Comment (onurctirtir) [inline src/test/regress/multi_schedule:112]: If the following change doesn't increase the time it takes to execute those three tests in total, let's parallelize them:

```suggestion
test: other_databases grant_role_from_non_maindb seclabel_non_maindb
```

File: src/test/regress/sql/seclabel_non_maindb.sql [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+SECURITY LABEL ON TABLE a IS 'citus_classified';
+
+\c regression
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('database1') ORDER BY node_type;
+
+-- Check that only the SECURITY LABEL for ROLES is propagated to the non-main databases on other nodes
+\c - - - :worker_1_port
+\c database_w1
+SELECT provider, objtype, label, objname FROM pg_seclabels ORDER BY objname;
+
```
Comment (onurctirtir) [inline src/test/regress/sql/seclabel_non_maindb.sql:47]: We already ensure this a few lines ago via the following:
```sql
-- Check the result
\c regression
SELECT node_type, result FROM get_citus_tests_label_provider_labels('user1') ORDER BY node_type;
SELECT node_type, result FROM get_citus_tests_label_provider_labels('"user 2"') ORDER BY node_type;
```

So can remove those lines:
```suggestion
```

**Review by onurctirtir — COMMENTED** [review]
1. Would be nice to add a few tests into metadata_sync_from_non_maindb.sql to test metadata-syncing code-path for seclabels. @gurkanindibay might help with what to test for metadata-syncing.

2. Given that this is a user-facing change, let's also add such a line to the PR description so that it can appear in the release changelog:

> DESCRIPTION: Adds support for distributed \`SECURITY LABEL on ROLE\` commands from the databases where Citus is not installed

File: src/test/regress/sql/seclabel_non_maindb.sql (hunk 3 above)
Comment (eaydingol) [reply src/test/regress/sql/seclabel_non_maindb.sql:47]: database_w1 does not have Citus extension, whereas regression database has.  In addition, at line 46 we retrieve all objects with security labels. That check ensures that we only propagate security labels associated with roles.

Comment (onurctirtir) [reply src/test/regress/sql/seclabel_non_maindb.sql:47]: I got it now 👍

And besides that, we can combine those two lines:
```sql
\c - - - :worker_1_port
\c database_w1
```

into

```sql
\c database_w1 - - :worker_1_port
```

to switch both the database and the server port at once.

Comment (eaydingol) [reply src/test/regress/sql/seclabel_non_maindb.sql]: Neat, updated it

File: src/test/regress/expected/metadata_sync_from_non_maindb.out [hunk 4]
```diff
@@ -121,6 +132,14 @@ select check_database_privileges('grant_role2pc''_user3','metadata_sync_2pc_db',
  (TEMPORARY,t)
 (12 rows)
 
+SELECT node_type, result FROM get_citus_tests_label_provider_labels('grant_role2pc_user4') ORDER BY node_type;
```
Comment (onurctirtir) [inline src/test/regress/expected/metadata_sync_from_non_maindb.out:135]: let's also check

```sql
SELECT node_type, result FROM get_citus_tests_label_provider_labels("grant_role2pc'_user1") ORDER BY node_type;
```

**Review by onurctirtir — APPROVED** [review]

File: src/test/regress/expected/metadata_sync_from_non_maindb.out (hunk 4 above)
Comment (eaydingol) [reply src/test/regress/expected/metadata_sync_from_non_maindb.out:142]: Added in the last commit, this requires special handling due to the single quote.

**Pull request merged**